### PR TITLE
[query] Update Graphite find calls to complete unterminated queries with all possible path values for **

### PR DESCRIPTION
**Pull request by robskillington**
**What this PR does / why we need it**:

Previously find calls did not special case `**` and such were looking for terminated and unterminated queries for fields that had any value for the first graphite path index (since matchers created for `**` are special and regex against the whole timeseries ID, they cannot be treated this way and need to complete all values for all possible path indexes).

This special cases the `**` case and completes tag values for all of the path indexes that could be possibly relevant to the find query.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

**Review by linasm — COMMENTED** [review]
Don't have much of a context for this code, but one part of the implementation looks somewhat suspicious (see the comments inline),.

File: src/query/graphite/storage/m3_wrapper.go [hunk 1]
```diff
@@ -92,16 +92,31 @@ func NewM3WrappedStorage(
...
+	StarStarUnterminatedTranslatedQuery
+)
+
 // TranslateQueryToMatchersWithTerminator converts a graphite query to tag
 // matcher pairs, and adds a terminator matcher to the end.
 func TranslateQueryToMatchersWithTerminator(
 	query string,
-) (models.Matchers, error) {
+) (models.Matchers, TranslatedQueryType, error) {
+	translatedQueryType := TerminatedTranslatedQuery
```
[the remark this answers is not in the document]
Comment (robskillington) [reply src/query/graphite/storage/m3_wrapper.go:113]: Sure thing.

File: src/query/api/v1/handler/graphite/find.go [hunk 2]
```diff
@@ -73,23 +73,33 @@ func mergeTags(
...
 
-	mapLength := len(terminatedResult.CompletedTags) + len(childResult.CompletedTags)
-	tagMap := make(map[string]nodeDescriptor, mapLength)
-
-	for _, tag := range terminatedResult.CompletedTags {
-		for _, value := range tag.Values {
-			descriptor := tagMap[string(value)]
-			descriptor.isLeaf = true
-			tagMap[string(value)] = descriptor
+	size := terminatedResultTags + childResultTags
```
[the remark this answers is not in the document]
Comment (robskillington) [reply src/query/api/v1/handler/graphite/find.go:96]: Fixed this, thanks Linas.

**Review by wesleyk — APPROVED** [review]
LGTM

**Pull request merged**